src/components/StoryLikeButton.tsx

```diff
@@ -16,20 +16,53 @@ const StoryLikeButtonFragment = graphql`
   }
 `;
 
+const StoryLikeButtonLikeMutation = graphql`
+  mutation StoryLikeButtonLikeMutation($id: ID!, $doesLike: Boolean!) {
+    likeStory(id: $id, doesLike: $doesLike) {
+      story {
+        ...StoryLikeButtonFragment
+      }
+    }
+  }
+`;
+
 export default function StoryLikeButton({ story }: Props): React.ReactElement {
   const data = useFragment<StoryLikeButtonFragment$key>(
     StoryLikeButtonFragment,
     story
   );
-  const onLikeButtonClicked = () => {
-    // To be filled in
-  };
+  const [commitMutation, isMutationInFlight] = useMutation(
+    StoryLikeButtonLikeMutation
+  );
+
+  function onLikeButtonClicked() {
+    commitMutation({
+      variables: {
+        id: data.id,
+        doesLike: !data.doesViewerLike,
+      },
+      optimisticUpdater: (store) => {
+        const fragment = graphql`
+          fragment StoryLikeButton_updatable on Story @updatable {
+            likeCount
+            doesViewerLike
+          }
+        `;
+        const { updatableData } = store.readUpdatableFragment(fragment, story);
+        const alreadyLikes = updatableData.doesViewerLike;
+        updatableData.doesViewerLike = !alreadyLikes;
+        updatableData.likeCount += alreadyLikes ? -1 : 1;
+      },
```

Comment:
instead can we get data from another query since relay will update matching objects ids